presto-memory/src/test/java/com/facebook/presto/plugin/memory/TestMemoryMetadata.java

```diff
@@ -170,6 +175,67 @@ public void testCreateSchema()
         assertEquals(metadata.listTables(SESSION, "default"), ImmutableList.of());
     }
 
+    @Test(expectedExceptions = PrestoException.class, expectedExceptionsMessageRegExp = "View already exists: test\\.test_view")
+    public void testCreateViewWithoutReplace()
+    {
+        SchemaTableName test = new SchemaTableName("test", "test_view");
+        try {
+            metadata.createView(SESSION, test, "test", false);
+        }
+        catch (Exception e) {
+            fail("should have succeeded");
+        }
+        metadata.createView(SESSION, test, "test", false);
+    }
+
+    @Test
+    public void testCreateViewWithReplace()
+    {
+        SchemaTableName test = new SchemaTableName("test", "test_view");
+
+        metadata.createView(SESSION, test, "aaa", true);
+        metadata.createView(SESSION, test, "bbb", true);
+
+        assertEquals(metadata.getViews(SESSION, test.toSchemaTablePrefix()).get(test).getViewData(), "bbb");
+    }
+
+    @Test
+    public void testViews()
+    {
+        SchemaTableName test1 = new SchemaTableName("test", "test_view1");
+        SchemaTableName test2 = new SchemaTableName("test", "test_view2");
+
+        // create views
+        metadata.createView(SESSION, test1, "test1", false);
+        metadata.createView(SESSION, test2, "test2", false);
+
+        // verify listing
+        List<SchemaTableName> list = metadata.listViews(SESSION, "test");
+        assertEqualsIgnoreOrder(list, ImmutableList.of(test1, test2));
+
+        // verify getting data
+        Map<SchemaTableName, ConnectorViewDefinition> views = metadata.getViews(SESSION, new SchemaTablePrefix("test"));
```

Comment:
Can you please also test partial or empty selection?

Reply:
We can add tests for
```java
// all schemas
new SchemaTablePrefix()

// exact match on schema and table
new SchemaTablePrefix("test", "test1")

// non-existent table
new SchemaTablePrefix("test", "nonexistingview")

// non-existent schema
new SchemaTablePrefix("nonexistingschema")